data-prepper-api/src/main/java/org/opensearch/dataprepper/model/codec/JsonDecoder.java

```diff
@@ -34,16 +41,15 @@ public void parse(InputStream inputStream, Instant timeReceived, Consumer<Record
         while (!jsonParser.isClosed() && jsonParser.nextToken() != JsonToken.END_OBJECT) {
             if (jsonParser.getCurrentToken() == JsonToken.START_ARRAY) {
                 parseRecordsArray(jsonParser, timeReceived, eventConsumer);
+            } else if (jsonParser.getCurrentToken() == JsonToken.START_OBJECT) {
```

Comment:
Will this break any existing functionality?

Thinking of, for example, reprocessing DLQ objects that are in this format. Wouldn't this change make it so this is read as one object instead of reading each Event from the json array?

```
{
  "dlqObjects" [ { object1, object2} ]
}
```

Reply:
This change only impacts two paths. Kafka buffer path and JsonInputCodec. Currently JsonInputCodec expects the input to have "[...]" If the input does not start with "[", it would fail. This new change would allow inputs with single json objects too. I don't think it would break anything. Let me know if you have clear example.